.pre-commit-config.yaml

```
@@ -0,0 +1,178 @@
+    rev: v8.18.2
+    hooks:
+      - id: gitleaks
+        name: "security: Detect hardcoded secrets"
+
+  # =================================================================== #
+  #                         Code Quality & Format                       #
+  # =================================================================== #
+
+  # ---------------------------- Python ----------------------------- #
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.4.4 # Using a version compatible with ruff-format
```

YAMLlint (1.37.1)

[warning] too few spaces before comment: expected 2

(comments)